assembly/NFT/NFT.ts

```diff
@@ -0,0 +1,229 @@
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function name(_: string): string {
+  return _name;
+}
+
+/**
+ * Return the NFT's symbol
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function symbol(_: string): string {
+  return _symbol;
+}
+
+/**
+ * Return the token URI (external link written in NFT where pictures or others are stored)
+ * @param {string} tokenId
+ * @return {string}
+ */
+export function tokenURI(tokenId: string): string {
+  if (Storage.has(baseURIKey)) {
+    return Storage.get(baseURIKey) + tokenId;
+  } else {
+    return '';
+  }
+}
+
+/**
+ * Return base URI
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function baseURI(_: string): string {
+  if (Storage.has(baseURIKey)) {
+    return Storage.get(baseURIKey);
+  } else {
+    return '';
+  }
+}
+
+/**
+ * Return the max supply possible
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function limitSupply(_: string): string {
+  return _maxSupply;
+}
+
+/**
+ * Return the current counter, if 10 NFT minted, returns '10'.
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function currentSupply(_: string): string {
+  if (Storage.has(counterKey)) {
+    return Storage.get(counterKey);
+  } else {
+    return '';
+  }
+}
+
+/**
+ * Return the tokenId's owner
+ * @param {string} tokenId
+ * @return {string}
+ */
+export function ownerOf(tokenId: string): string {
+  if (Storage.has(ownerTokenKey + tokenId.toString())) {
+    return Storage.get(ownerTokenKey + tokenId.toString());
+  } else {
+    return '';
+  }
+}
+
+// ==================================================== //
+// ====                 TRANSFER                   ==== //
+// ==================================================== //
+
+/**
+ * The to address becomes the owner of the next token (if current tokenID = 10, will mint 11 )
+ * Check if max supply is not reached
+ * @param {string} args - byte string containing an owner's account (Address).
+ * @return {string}
+ */
+export function mint(args: string): string {
+  if (u32(parseInt(limitSupply(''))) > u32(parseInt(currentSupply('')))) {
+    const addr = Address.fromByteString(args);
+    _increment('');
+    const tokenID: string = currentSupply('');
+    const key = ownerTokenKey + tokenID;
+    Storage.set(key, addr.toByteString());
+    generateEvent(`tokenId ${tokenID} minted to ${addr.toByteString()} `);
+  } else {
+    generateEvent(`Max supply reached`);
+  }
+  return '';
+}
+
+/**
+ * Increment the NFT counter
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+function _increment(_: string): string {
```

Comment:
i read the issue here: https://github.com/massalabs/massa-sc-std/issues/18
But doest it really apply here ?? i mean its an internal function of the contract. it will not be called from the outside

Reply:
good catch, indeed this function is not exported